packages/events/src/service/events/events.ts

```diff
@@ -389,7 +389,15 @@ export async function addAction(
 
   const updatedEvent = await getEventById(eventId)
 
-  await indexEvent(updatedEvent, configuration)
+  const isDraft =
+    getStatusFromActions(updatedEvent.actions) === EventStatus.enum.CREATED
+
+  // Only send the event to Elasticsearch if it is not a draft and action status is accepted
+  const shouldIndexEvent = !isDraft && status === ActionStatus.Accepted
+
+  if (shouldIndexEvent) {
+    await indexEvent(updatedEvent, configuration)
+  }
```

Comment:
@makelicious We had to update this because of the way write actions are saved now(for each write action, there is a `requested` and an `accepted` action.

Reply:
had to change this approach after we found out (after a test from `event.index.flag.test.ts` broke) we can't exclude `Requested` action to be not indexed for some action e.g - `Waiting external validation action`